### PR TITLE
Gate library and example for synqs fermion backend

### PR DESCRIPTION
**Pull request by RohitPrasadBhatt**
This allows for using our fermion backend.

### CODE REVIEW

File: qiskit_cold_atom/synqs_fermions/fqs_gate_library.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+        super().__init__(name="int", num_qubits=len(wires), params=[theta_u], label=None)
+
+@add_gate
+def inter(self, wires:List[int], theta_u:float):
+    # pylint: disable=invalid-name
+    """add the inter gate to a QuantumCircuit"""
+    return self.append(InterGate(wires,theta_u), wires)
+
+class HopGate(Gate):
```
Comment (eggerdj) [inline qiskit_cold_atom/synqs_fermions/fqs_gate_library.py:44]: How does this relate to the existing fermion gates https://github.com/Qiskit-Extensions/qiskit-cold-atom/blob/410332a470ad371eab0cb74c2b88ee39aae1b887/qiskit_cold_atom/fermions/fermion_gate_library.py#L342 ? We should try and use those as much as possible.

File: qiskit_cold_atom/synqs_fermions/__init__.py [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# Gate imports
+from .fqs_gate_library import (
```
Comment (eggerdj) [inline qiskit_cold_atom/synqs_fermions/__init__.py:2]: We should use the existing gates as much as possible and if something is lacking from those we should extend the library.

**Review by eggerdj — CHANGES_REQUESTED** [review]
Looks like this PR should be combined with https://github.com/Qiskit-Extensions/qiskit-cold-atom/pull/13 so that we have the choice over the backend.

**Comment by fretchen** [comment]
I think that this one can be closed as it is now superseeded by #13

**Comment by fretchen** [comment]
I pushed the discussed changes. However, one important problem remains. QISKIT is not working with the interleaved notation, while other frameworks and our back-end do. @RohitPrasadBhatt therefore suggested to create a flag in the config that allows to notify the backend which kind of notation it receives for fermionic devices and also to say which kind of notation it expects. I fully support this idea. Should this be an extra issue ?

**Pull request merged**